jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,735 @@
+                return false;
+
+            for (Predicate<Request> predicate : _predicates)
+            {
+                if (predicate.test(request))
+                    return true;
+            }
+            return false;
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests the {@link Connector#getName() name} of the
+     * {@link ConnectionMetaData#getConnector() connector} obtained from {@link Request#getConnectionMetaData()}
+     */
+    public static class ConnectorPredicate implements Predicate<Request>
+    {
+        private final String _connector;
+
+        public ConnectorPredicate(String connector)
+        {
+            this._connector = Objects.requireNonNull(connector);
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _connector.equals(request.getConnectionMetaData().getConnector().getName());
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s@%x{%s}", getClass().getSimpleName(), hashCode(), _connector);
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests an {@link InetAddressPattern}
+     * against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     * {@link Request#getConnectionMetaData()}.
+     */
+    public static class InetAddressPatternPredicate implements Predicate<Request>
+    {
+        public static InetAddress getInetAddress(SocketAddress socketAddress)
+        {
+            if (socketAddress instanceof InetSocketAddress inetSocketAddress)
+            {
+                if (inetSocketAddress.isUnresolved())
+                {
+                    try
+                    {
+                        return InetAddress.getByName(inetSocketAddress.getHostString());
+                    }
+                    catch (UnknownHostException e)
+                    {
+                        if (LOG.isTraceEnabled())
+                            LOG.trace("ignored", e);
+                        return null;
+                    }
+                }
+
+                return inetSocketAddress.getAddress();
+            }
+            return null;
+        }
+
+        private final InetAddressPattern _pattern;
+
+        public InetAddressPatternPredicate(InetAddressPattern pattern)
+        {
+            _pattern = pattern;
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _pattern.test(getInetAddress(request.getConnectionMetaData().getRemoteSocketAddress()));
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return _pattern.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            return obj instanceof InetAddressPatternPredicate inetAddressPatternPredicate && _pattern.equals(inetAddressPatternPredicate._pattern);
+        }
```

Comment:
Why are these methods overridden, since they are not for all the other predicates?
If not necessary, remove them, or else they should be present in all predicates.

Reply:
They are needed because the InetPredicates are put into a Set.     So I guess all the predicates should have them.